addons/calendar/controllers/main.py

```diff
@@ -1,49 +1,45 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
 import json
-import openerp
-import openerp.http as http
-from openerp.http import request
-import openerp.addons.web.controllers.main as webmain
-import json
+
+import odoo
+from odoo import http, registry
+from odoo.http import request
 
 
-class meeting_invitation(http.Controller):
+class MeetingInvitation(http.Controller):
 
     @http.route('/calendar/meeting/accept', type='http', auth="calendar")
     def accept(self, db, token, action, id, **kwargs):
-        registry = openerp.modules.registry.RegistryManager.get(db)
-        attendee_pool = registry.get('calendar.attendee')
-        with registry.cursor() as cr:
-            attendee_id = attendee_pool.search(cr, openerp.SUPERUSER_ID, [('access_token', '=', token), ('state', '!=', 'accepted')])
-            if attendee_id:
-                attendee_pool.do_accept(cr, openerp.SUPERUSER_ID, attendee_id)
+        with registry(db).cursor() as cr:
+            env = request.env(cr, odoo.SUPERUSER_ID)
+            attendee = env['calendar.attendee'].search([('access_token', '=', token), ('state', '!=', 'accepted')])
+            if attendee:
+                attendee.do_accept()
         return self.view(db, token, action, id, view='form')
 
     @http.route('/calendar/meeting/decline', type='http', auth="calendar")
     def declined(self, db, token, action, id):
-        registry = openerp.modules.registry.RegistryManager.get(db)
-        attendee_pool = registry.get('calendar.attendee')
-        with registry.cursor() as cr:
-            attendee_id = attendee_pool.search(cr, openerp.SUPERUSER_ID, [('access_token', '=', token), ('state', '!=', 'declined')])
-            if attendee_id:
-                attendee_pool.do_decline(cr, openerp.SUPERUSER_ID, attendee_id)
+        with registry(db).cursor() as cr:
+            env = request.env(cr, odoo.SUPERUSER_ID)
```

Comment:
Don't use request.env instead create new Environment here(for context you may fetch the lang of partner of attendee).